SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialConnectionManager.cpp

```diff
@@ -18,6 +18,24 @@ DEFINE_LOG_CATEGORY(LogSpatialConnectionManager);
 
 using namespace SpatialGDK;
 
+namespace AsyncUtil
+{
+template <typename FN>
+void AsyncTaskGameThreadOutsideGC(FN&& Func) // A little wrapper which ensures the task is run outside GC
```

Comment:
Are you sure this won't spam a lot while we wait for GC to finish?

Reply:
E.g. should we consider a delay?

Reply:
imho we should only create UObjects on the game thread.

Reply:
But this is on the game thread?

Reply:
I think GC is started and completed between frames so it should be good on the next tick I think (this code is actually wrong but replacing it with something similar to kick off on the next frame)

Reply:
I should have been more specific - whilst it is running in an async task on the game thread, it isn't scheduled to run at a particular time, e.g. tied to an existing tick function, and therefore it can run during GC?

I would prefer it to be properly scheduled and/or the object new'ed from within an existing game thread tick function.

Reply:
Yeah I think I agree with this in general, pretty sure it's unintentional (ofc it's undocumented) because the engine also creates objects in ::GameThread tasks

Thoughts on this as a quick 0.13 patch and maybe we could create a ticket to rework the code?